src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/personalrelationships/profiledetails/ContactPersonProfileDetailsReconciliationService.kt

```diff
@@ -29,18 +32,23 @@ import uk.gov.justice.digital.hmpps.prisonertonomisupdate.personalrelationships.
 import uk.gov.justice.digital.hmpps.prisonertonomisupdate.profiledetails.ProfileDetailsNomisApiService
 import uk.gov.justice.digital.hmpps.prisonertonomisupdate.services.NomisApiService
 import uk.gov.justice.digital.hmpps.prisonertonomisupdate.services.doApiCallWithRetries
+import kotlin.concurrent.atomics.AtomicInt
+import kotlin.concurrent.atomics.ExperimentalAtomicApi
+import kotlin.math.ceil
 
 typealias Mismatches = List<String>
 
-@OptIn(ExperimentalCoroutinesApi::class)
+@OptIn(ExperimentalCoroutinesApi::class, ExperimentalAtomicApi::class)
 @Service
 class ContactPersonProfileDetailsReconciliationService(
   @Autowired private val nomisIdsApi: NomisApiService,
   @Autowired private val nomisApi: ProfileDetailsNomisApiService,
   @Autowired private val dpsApi: ContactPersonProfileDetailsDpsApiService,
   @Value("\${reports.contact-person.profile-details.reconciliation.page-size:20}") private val pageSize: Long = 20,
+  @Value("\${reports.contact-person.profile-details.reconciliation.parallel-jobs:20}") private val parallelJobs: Long = 20,
   @Value("\${feature.recon.contact-person.profile-details:true}") private val reconciliationTurnedOn: Boolean = true,
   @Autowired private val telemetryClient: TelemetryClient,
+  @Autowired(required = false) private val channelActivityDebugger: ContactPersonProfileDetailsChannelActivityDebugger? = null,
```

Comment:
This debugger monitors the channel population and number of parallel checks in progress and is not created by default. It's intended for use locally when trying to guesstimate page size and number of parallel jobs, but could be turned on in a real env. to refine these values.